onnxruntime/python/onnxruntime_pybind_state.cc

```diff
@@ -654,6 +655,13 @@ static void RegisterExecutionProviders(InferenceSession* sess, const std::vector
     } else if (type == kRknpuExecutionProvider) {
 #ifdef USE_RKNPU
       RegisterExecutionProvider(sess, *onnxruntime::CreateExecutionProviderFactory_Rknpu());
+#endif
+    } else if (type == kCoreMLExecutionProvider) {
+#if defined(USE_COREML)
+#if !defined(__APPLE__)
```

Comment:
I think `USE_COREML` requires `__APPLE__` already (if we use build.py).
https://github.com/microsoft/onnxruntime/blob/002e427c5bd0d60d765775382b6fd24bc4a76db5/tools/ci_build/build.py#L880-L881
should this test for non-iOS instead?
another note, unrelated to this PR, the "macOS" referred to in build.py should probably be "Apple platform" or something.

Reply:
Yes. You are right. Will update.